modules/pool/lib/pool-on-chain-tokenpair-data.ts

```diff
@@ -150,29 +151,53 @@ function generateTokenPairs(filteredPools: PoolInput[]): TokenPair[] {
         // create all pairs for pool
         for (let i = 0; i < pool.tokens.length - 1; i++) {
             for (let j = i + 1; j < pool.tokens.length; j++) {
-                //skip pairs with phantom BPT
-                if (pool.tokens[i].address === pool.address || pool.tokens[j].address === pool.address) continue;
```

Comment:
We should only add BPTs for CSP pools, as the others dont support those swaps, do they?

Reply:
I believe that any pool that contains its BPT as a token should support those swaps, but now that you're asking, it might be interesting to double check that assumption 😅